### DISCUSSION TITLE
Almost Amazon- error: cannot read property of 'forEach' of undefined

### DISCUSSION THREAD

**Discussion by nashvegasnate** [nss-evening-cohort-14/evening-client-side#155] [Help Me!] [answered]
I have not been successful in getting the addAuthorForm to work, so there may be a connection between these two issues. My eslint is showing no errors, however, my console is showing the following error:
   TypeError: Cannot read property 'forEach' of undefined
    at showAuthors (authors.js?39be:7)
    at startApp (startApp.js?a45b:19)
    at Object.eval [as next] (router.js?f5e9:19)

I have tried to fix the issue in showAuthors and at startApp, however, nothing seems to work. 

Below are the three links to searches on the issue:
https://support.bigcommerce.com/s/question/0D54O00006h3AvCSAU/uncaught-typeerror-cannot-read-property-foreach-of-undefined-when-using-the-array-widget-ui-type
https://stackoverflow.com/questions/38908243/cannot-read-property-foreach-of-undefined
https://www.reddit.com/r/learnjavascript/comments/7a9iib/uncaught_typeerror_cannot_read_property_foreach/

Here is the link to my repo:
https://github.com/nashvegasnate/IN-CLASS-almost-amazon-template

**Comment by drteresavasquez** [comment]
If your console is showing errors, I'm betting your terminal has some too. Does it say it compiled successfully?

**Comment by nashvegasnate** [comment]
It does say it has compiled successfully. That is why it is so confusing.

**Reply by drteresavasquez** [reply]
I'm sure you've tried this but stop and start your server and then look for everywhere you are calling showAuthors. Make sure your import statements are correct and that you are passing it an array. I could not tell everywhere you are calling it since I tried looking through your code but I need direct links to those to make sure I see everywhere you are calling it.

**Comment by drteresavasquez** [comment, marked as answer]
@nashvegasnate oh I think I see it. In this file, https://github.com/nashvegasnate/IN-CLASS-almost-amazon-template/blob/main/src/javascripts/views/startApp.js#L19 you're calling showAuthors twice instead of `getAuthors().then()...`

You're getting that error because showAuthors expects an array and you're not passing it anything because it's the wrong function call.

**Reply by nashvegasnate** [reply]
That got rid of the error! I guess I couldn't see the forest for the trees. Still not able to add an author, but working on that today. Thanks for the assist!

**Reply by drteresavasquez** [reply]
Of course!
